flang/lib/Optimizer/Transforms/ArrayValueCopy.cpp

```diff
@@ -353,24 +339,30 @@ void ArrayCopyAnalysis::arrayMentions(
 
   // Process the worklist until done.
   while (!queue.empty()) {
-    auto *operand = queue.pop_back_val();
-    auto *owner = operand->getOwner();
-    if (!owner)
-      continue;
+    mlir::OpOperand *operand = queue.pop_back_val();
+    mlir::Operation *owner = operand->getOwner();
     auto structuredLoop = [&](auto ro) {
       if (auto blockArg = ro.iterArgToBlockArg(operand->get())) {
-        auto arg = blockArg.getArgNumber();
-        auto output = ro.getResult(ro.finalValue() ? arg : arg - 1);
+        int64_t arg = blockArg.getArgNumber();
+        mlir::Value output = ro.getResult(ro.finalValue() ? arg : arg - 1);
         appendToQueue(output);
         appendToQueue(blockArg);
       }
     };
-    auto branchOp = [&](mlir::Block *dest, auto operands) {
-      for (auto i : llvm::enumerate(operands))
-        if (operand->get() == i.value()) {
-          auto blockArg = dest->getArgument(i.index());
-          appendToQueue(blockArg);
-        }
+    auto branchOp = [&](mlir::Block *dest, mlir::OperandRange operands) {
+      if (operands.empty())
+        return;
+
+      // Check if this operand is within the range.
+      unsigned operandIndex = operand->getOperandNumber();
```

Comment:
Are these returning `unsigned` or `size_t`? (It's not always consistent, so this could be an implicit truncation.)

Reply:
`unsigned` in this case.